spec/mailers/notifier_spec.rb

```diff
@@ -44,4 +44,84 @@
     end
 
   end
+
+
+  describe "new crop request" do
+    let(:member) { FactoryGirl.create(:crop_wrangling_member) }
+    let(:crop) { FactoryGirl.create(:crop_request) }
+    let(:mail) { Notifier.new_crop_request(member, crop) }
+
+    it 'sets the subject correctly' do
+      mail.subject.should == "New crop request"
+    end
+
+    it 'comes from noreply@growstuff.org' do
+      mail.from.should == ['noreply@growstuff.org']
+    end
+
+    it 'sends the mail to the recipient of the notification' do
+      mail.to.should == [member.email]
+    end
+
+    it 'includes the requested crop URL' do
+      mail.body.encoded.should match crop_url(crop)
+    end
+  end
+
+  describe "crop approved" do
+    let(:member) { FactoryGirl.create(:member) }
+    let(:crop) { FactoryGirl.create(:crop) }
+    let(:mail) { Notifier.crop_request_approved(member, crop) }
+
+    it 'sets the subject correctly' do
+      expect(mail.subject).to eq "Magic bean has been approved"
+    end
+
+    it 'comes from noreply@growstuff.org' do
+      expect(mail.from).to eq ['noreply@growstuff.org']
+    end
+
+    it 'sends the mail to the recipient of the notification' do
+      expect(mail.to).to eq [member.email]
+    end
+
+    it 'includes the approved crop URL' do
+      expect(mail.body.encoded).to match crop_url(crop)
+    end
+
+    it 'includes links to plant, harvest and stash seeds for the new crop' do
+      expect(mail.body.encoded).to match new_planting_url
+      expect(mail.body.encoded).to match new_harvest_url
+      expect(mail.body.encoded).to match new_seed_url
```

Comment:
How can I check that the query parameter is included as well? The following wasn't working for some reason:

`expect(mail.body.encoded).to match new_planting_url(crop_id: crop.id)`

Can we just use `have_link` in this test?